go/cable.go

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright 2024 ScopeDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package scopedb
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"math"
+	"time"
+
+	"github.com/apache/arrow/go/v17/arrow"
+)
+
+type ArrowBatchCable struct {
+	c *Client
+
+	schema      *arrow.Schema
+	transforms  string
+	currentSize uint64
+	sendBatches []*arrowSendRecord
+	sendBatchCh chan *arrowSendRecord
+
+	BatchSize     uint64
+	BatchInterval time.Duration
+}
+
+type arrowSendRecord struct {
+	record arrow.Record
+	err    chan error
+}
+
+func (c *Client) ArrowBatchCable(schema *arrow.Schema, transforms string) *ArrowBatchCable {
+	cable := &ArrowBatchCable{
+		c:             c,
+		schema:        schema,
+		transforms:    transforms,
+		currentSize:   0,
+		sendBatches:   make([]*arrowSendRecord, 0),
+		sendBatchCh:   make(chan *arrowSendRecord),
+		BatchSize:     1024 * 1024, // default to 1MiB
```

Comment:
We can make it larger, for example 16MB.

Reply:
Updated.